packages/rum/src/domain/record/record.spec.ts

```diff
@@ -422,6 +422,23 @@ describe('record', () => {
     })
   })
 
+  describe('it should not record when full snapshot is pending', () => {
+    it('ignores any record while a full snapshot is pending', () => {
+      lifeCycle = new LifeCycle()
+      recordApi = record({
+        emit: emitSpy,
+        configuration: { defaultPrivacyLevel: DefaultPrivacyLevel.ALLOW } as RumConfiguration,
+        lifeCycle,
+        viewContexts: {
+          findView: () => ({ id: FAKE_VIEW_ID, startClocks: {} }),
+        } as any,
+        defineSnapshotPending: true,
+      })
+      const records = getEmittedRecords()
+      expect(records.length).toBe(0)
```

Comment:
**💬 suggestion:** ‏instead of having a `defineSnapshotPending` that is only used for test, you could induce the pending state by mocking the idle callback then triggering a view change, something like:


```ts
mockRequestIdleCallback()
startRecording()
newView()

// this normally creates a "focus" record
window.dispatchEvent(createNewEvent('focus'))

expect(emitSpy).not.toHaveBeenCalled()
```